### PR TITLE
Fix line numbers and search related bugs

### PR DESCRIPTION
**Pull request by arijit79**
This PR fixes two high severity bugs within `minus`
- Duplicity of  line numbers on appending str
   This occurred because text takes the last line pushed into account from `PagerState::formatted_lines` which is as in the name itself formatted causing the line numbers to be displayed twice

- Pressing `n` does not takes to the next match
   This occurred because `set_match_indices`  was sort of buggy with the new implementation of appending text

### CODE REVIEW

File: src/core/ev_handler.rs [hunk 1]
```diff
@@ -101,7 +103,7 @@ pub fn handle_event(
             }
             // Decrement the s_mark and get the preceeding index
             p.search_mark = p.search_mark.saturating_sub(1);
-            let y = p.search_idx[p.search_mark];
+            let y = *p.search_idx.iter().nth(p.search_mark).unwrap();
```
Comment (iandwelker) [inline src/core/ev_handler.rs:106]: Why did you change this? Isn't this functionally the same as before, just syntactically more complex?

File: src/lib.rs [hunk 2]
```diff
@@ -677,17 +732,22 @@ impl PagerState {
...
                     idx,
+                    #[cfg(feature = "search")]
+                    &mut search_idx,
                 )
             })
             .collect::<Vec<String>>();
 
+        #[cfg(feature = "search")]
+        {
+            self.search_idx.append(&mut search_idx);
```
Comment (iandwelker) [inline src/lib.rs:743]: Wouldn't you just want to do `self.search_idx = search_idx`? If you change the search term, wouldn't this cause the matches from both of the searches to be contained in the search_idx?

File: src/core/ev_handler.rs (hunk 1 above)
Comment (arijit79) [reply src/core/ev_handler.rs:106]: `PagerState::search_idx` is now a BTreeSet and hence to extract things from it, we just have to do this complex thing.

File: src/lib.rs (hunk 2 above)
Comment (arijit79) [reply src/lib.rs:743]: Oh! I guess I did it by mistake. I will fix it in the morning.

File: src/core/ev_handler.rs (hunk 1 above)
Comment (iandwelker) [reply src/core/ev_handler.rs:106]: Ah, gotcha; that makes sense.

File: src/lib.rs (hunk 2 above)
Comment (iandwelker) [reply src/lib.rs]: [9404e55](https://github.com/arijit79/minus/pull/62/commits/9404e55974b7aaf44d0adc5c55987763a72d8937) resolves this.
